### ISSUE TITLE
Unable to F5 .NET Aspire Started Application from VS

### ISSUE THREAD

**Issue by LarsPal** [dotnet/aspire#6609]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

I created a brand new .NET Aspire Starter App in VS, but when I hit F5 I get this exception..

`Unhandled exception. System.AggregateException: One or more errors occurred. (Failed to configure dashboard resource because ASPNETCORE_URLS environment variable was not set.)
 ---> Microsoft.Extensions.Options.OptionsValidationException: Failed to configure dashboard resource because ASPNETCORE_URLS environment variable was not set.`
![Image](https://github.com/user-attachments/assets/ce456f19-aa21-4a5d-9e0e-195e27c7ca64)

If I start it with CLI or add ASPNETCORE_URLS  in launchSettings.json it starts a expected. Only problem is that the Aspire Dashboard isn't opened in the browser automatically.
![Image](https://github.com/user-attachments/assets/efaad8a9-6b63-4355-b08b-2be7805bfa41)

If I change to `"Aspire.Hosting.Dcp": "Debug" ` I get this warning
![Image](https://github.com/user-attachments/assets/bf6e3d88-7096-450d-9d60-105c588a16cc)
Could that be an indication what is wrong?

<details open>
  <summary>dotnet --info</summary>
.NET SDK:
 Version:           8.0.403
 Commit:            c64aa40a71
 Workload version:  8.0.400-manifests.3996aa5e
 MSBuild version:   17.11.9+a69bbaaf5

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.403\

.NET workloads installed:
Configured to use loose manifests when installing new manifests.
 [aspire]
   Installation Source: SDK 8.0.400
   Manifest Version:    8.1.0/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.aspire\8.1.0\WorkloadManifest.json
   Install Type:        FileBased

Host:
  Version:      8.0.10
  Architecture: x64
  Commit:       81cabf2857/

.NET SDKs installed:
  7.0.317 [C:\Program Files\dotnet\sdk]
  8.0.303 [C:\Program Files\dotnet\sdk]
  8.0.403 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.35 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.35 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  C:\Repos\TetraPak.ServiceTool.Server\global.json

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</details>

Here is another issue that seems very related. [4046](https://github.com/dotnet/aspire/issues/4046)

I can also add that I have been able to run Aspire Applications without any problem. But suddenly a couple of days ago I started to encounter this exception without me knowing what might have changed on my computer.

**Comment by joperezr** [comment]
[Triage] -  Thanks for logging this issue. Can you please share a repro repository of the issue so we can investigate further?

**Comment by joperezr** [comment]
Also, could you try updating aspire 8.2.2 by running `dotnet workload update` from the command line to see if the issue still repros?

**Comment by LarsPal** [comment]
It's a '.NET Aspire Starter App' solution.
[AspireApp1.zip](https://github.com/user-attachments/files/17643082/AspireApp1.zip)
This is how the launchSettings.json looks like:
```
{
  "$schema": "https://json.schemastore.org/launchsettings.json",
  "profiles": {
    "https": {
      "commandName": "Project",
      "dotnetRunMessages": true,
      "launchBrowser": true,
      "applicationUrl": "https://localhost:17278;http://localhost:15202",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development",
        "DOTNET_ENVIRONMENT": "Development",
        "DOTNET_DASHBOARD_OTLP_ENDPOINT_URL": "https://localhost:21027",
        "DOTNET_RESOURCE_SERVICE_ENDPOINT_URL": "https://localhost:22200"
      }
    },
    "http": {
      "commandName": "Project",
      "dotnetRunMessages": true,
      "launchBrowser": true,
      "applicationUrl": "http://localhost:15202",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development",
        "DOTNET_ENVIRONMENT": "Development",
        "DOTNET_DASHBOARD_OTLP_ENDPOINT_URL": "http://localhost:19090",
        "DOTNET_RESOURCE_SERVICE_ENDPOINT_URL": "http://localhost:20013"
      }
    }
  }
}
```
If I add `"ASPNETCORE_URLS": "https://localhost:17278;http://localhost:15202"` to launchSettings.json. Then I won't get the exception but the Aspire Dashboard isn't opened in the browser automatically.
This is the output then:
```
info: Aspire.Hosting.DistributedApplication[0]
      Aspire version: 8.1.0+d304c5f6f15bcd4f34f1841b33870cfab88e6937
info: Aspire.Hosting.DistributedApplication[0]
      Distributed application starting.
info: Aspire.Hosting.DistributedApplication[0]
      Application host directory is: C:\Users\sepalssonl\source\repos\AspireApp1\AspireApp1.AppHost
info: Aspire.Hosting.Dcp.DcpHostService[0]
      Starting DCP with arguments: start-apiserver --monitor 18460 --detach --kubeconfig "C:\Users\sepalssonl\AppData\Local\Temp\aspire.eh02bd3q.n0h\kubeconfig"
dbug: Aspire.Hosting.Dcp.ApplicationExecutor[0]
      Watching over DCP Container resources.
dbug: Aspire.Hosting.Dcp.ApplicationExecutor[0]
      Watching over DCP Executable resources.
dbug: Aspire.Hosting.Dcp.ApplicationExecutor[0]
      Watching over DCP Endpoint resources.
dbug: Aspire.Hosting.Dcp.ApplicationExecutor[0]
      Watching over DCP Service resources.
info: Aspire.Hosting.Dcp.start-apiserver.api-server[0]
      Starting API server...
info: Aspire.Hosting.Dcp.start-apiserver.api-server[0]
      API server started        {"Address": "::1", "Port": 59356}
info: Aspire.Hosting.Dcp.start-apiserver.dcp-host[0]
      Starting DCP controller host
info: Aspire.Hosting.Dcp.start-apiserver.dcp-host[0]
      Started all services      {"count": 1}
dbug: Aspire.Hosting.Dcp.KubernetesService[0]
      Successfully read Kubernetes configuration from 'C:\Users\sepalssonl\AppData\Local\Temp\aspire.eh02bd3q.n0h\kubeconfig' after 2218 milliseconds.
info: Aspire.Hosting.Dcp.dcpctrl.IdeExecutableRunner[0]
      Executables cannot be started via IDE: missing required environment variable 'DEBUG_SESSION_PORT'
info: Aspire.Hosting.Dcp.dcpctrl[0]
      starting controller manager
info: Aspire.Hosting.DistributedApplication[0]
      Now listening on: https://localhost:17278
info: Aspire.Hosting.DistributedApplication[0]
      Login to the dashboard at https://localhost:17278/login?t=e2f5879d60421ccafe125d6bfaa4c68e
info: Aspire.Hosting.Dcp.dcpctrl.ExecutableReconciler[0]
      starting process...       {"Executable": {"name":"apiservice-fa4tb0nndo"}, "Reconciliation": 2, "executable": "C:\\Program Files\\dotnet\\dotnet.exe"}
info: Aspire.Hosting.DistributedApplication[0]
      Distributed application started. Press Ctrl+C to shut down.
info: Aspire.Hosting.Dcp.dcpctrl.ExecutableReconciler[0]
      process started   {"Executable": {"name":"apiservice-fa4tb0nndo"}, "Reconciliation": 2, "executable": "C:\\Program Files\\dotnet\\dotnet.exe", "PID": 13344}
info: Aspire.Hosting.Dcp.dcpctrl.ServiceReconciler[0]
      service /apiservice is now in state Ready {"ServiceName": {"name":"apiservice"}, "Reconciliation": 7}
info: Aspire.Hosting.Dcp.dcpctrl.ExecutableReconciler[0]
      starting process...       {"Executable": {"name":"webfrontend-e42kjkc74g"}, "Reconciliation": 6, "executable": "C:\\Program Files\\dotnet\\dotnet.exe"}
info: Aspire.Hosting.Dcp.dcpctrl.ExecutableReconciler[0]
      process started   {"Executable": {"name":"webfrontend-e42kjkc74g"}, "Reconciliation": 6, "executable": "C:\\Program Files\\dotnet\\dotnet.exe", "PID": 6152}
info: Aspire.Hosting.Dcp.dcpctrl.ServiceReconciler[0]
      service /webfrontend is now in state Ready        {"ServiceName": {"name":"webfrontend"}, "Reconciliation": 9}
info: Aspire.Hosting.Dcp.dcpctrl.ExecutableReconciler[0]
      starting process...       {"Executable": {"name":"aspire-dashboard-sngfsjam"}, "Reconciliation": 7, "executable": "C:\\Program Files\\dotnet\\packs\\Aspire.Dashboard.Sdk.win-x64\\8.1.0\\tools\\Aspire.Dashboard.exe"}
info: Aspire.Hosting.Dcp.dcpctrl.ExecutableReconciler[0]
      process started   {"Executable": {"name":"aspire-dashboard-sngfsjam"}, "Reconciliation": 7, "executable": "C:\\Program Files\\dotnet\\packs\\Aspire.Dashboard.Sdk.win-x64\\8.1.0\\tools\\Aspire.Dashboard.exe", "PID": 15212}
dbug: Aspire.Hosting.Dcp.ApplicationExecutor[0]
      Starting log streaming for aspire-dashboard-sngfsjam
```

**Comment by LarsPal** [comment]
I tried to update to Aspire 8.2.2. It seems to be working but it is not updated.
```
Microsoft Windows [Version 10.0.19045.5011]
(c) Microsoft Corporation. All rights reserved.

C:\WINDOWS\system32>cd C:\Users\sepalssonl\source\repos\AspireApp1

C:\Users\sepalssonl\source\repos\AspireApp1>dotnet workload update

Updated advertising manifest microsoft.net.sdk.maccatalyst.
Updated advertising manifest microsoft.net.workload.emscripten.current.
Updated advertising manifest microsoft.net.workload.emscripten.net7.
Updated advertising manifest microsoft.net.sdk.macos.
Updated advertising manifest microsoft.net.sdk.ios.
Updated advertising manifest microsoft.net.workload.emscripten.net6.
Updated advertising manifest microsoft.net.sdk.android.
Updated advertising manifest microsoft.net.workload.mono.toolchain.current.
Updated advertising manifest microsoft.net.workload.mono.toolchain.net7.
Updated advertising manifest microsoft.net.sdk.tvos.
Updated advertising manifest microsoft.net.workload.mono.toolchain.net6.
Updated advertising manifest microsoft.net.sdk.maui.
Updated advertising manifest microsoft.net.sdk.aspire.

Successfully updated workload(s): aspire.


C:\Users\sepalssonl\source\repos\AspireApp1>dotnet workload install aspire

Workload(s) 'aspire' are already installed.

Successfully installed workload(s) .


C:\Users\sepalssonl\source\repos\AspireApp1>dotnet workload list

Installed Workload Id      Manifest Version      Installation Source
--------------------------------------------------------------------
aspire                     8.1.0/8.0.100         SDK 8.0.400

Use `dotnet workload search` to find additional workloads to install.

Updates are available for the following workload(s): aspire. Run `dotnet workload update` to get the latest.
```

**Comment by davidfowl** [comment]
There's something off with Visual Studio:

```
Executables cannot be started via IDE: missing required environment variable 'DEBUG_SESSION_PORT'
```

This log is the signal that the IDE isn't setting up the right state. Can you open up your VS installer, click Modify -> Individual Components, then search for Aspire and see if it's checked.

**Comment by LarsPal** [comment]
Individual Components -> Aspire wasn't checked and this was a conscious decision by our team. Because we install Aspire with a InstallAspireWorkload.bat file. The file looks like this:
```
dotnet workload clean --all
dotnet workload update --source https://api.nuget.org/v3/index.json --from-rollback-file .\aspire-rollback.txt
dotnet workload install aspire --source https://api.nuget.org/v3/index.json --from-rollback-file .\aspire-rollback.txt
```
The aspire-rollback.txt have this content.
```
{
 "microsoft.net.sdk.aspire": "8.1.0/8.0.100"
}
```
This was a way to make sure all developers would have the same version of Aspire. Our understanding was that if you check 'Individual Components -> Aspire' in VS Installer then `dotnet workload clean --all` isn't able to clean/remove it. Maybe we misunderstood something.


Now I have checked 'Individual Components -> Aspire' in the VS Installer and installed Aspire from VS Installer. I restarted my computer after that. Then I tried to update Aspire to 8.2.2 again but unfortunately with the same result.
```
C:\Users\sepalssonl\source\repos\AspireApp1>dotnet workload update

Updated advertising manifest microsoft.net.workload.emscripten.net6.
Updated advertising manifest microsoft.net.sdk.macos.
Updated advertising manifest microsoft.net.workload.emscripten.current.
Updated advertising manifest microsoft.net.sdk.aspire.
Updated advertising manifest microsoft.net.workload.mono.toolchain.net7.
Updated advertising manifest microsoft.net.sdk.maccatalyst.
Updated advertising manifest microsoft.net.workload.mono.toolchain.current.
Updated advertising manifest microsoft.net.sdk.ios.
Updated advertising manifest microsoft.net.sdk.maui.
Updated advertising manifest microsoft.net.sdk.tvos.
Updated advertising manifest microsoft.net.sdk.android.
Updated advertising manifest microsoft.net.workload.emscripten.net7.
Updated advertising manifest microsoft.net.workload.mono.toolchain.net6.

Successfully updated workload(s): aspire.


C:\Users\sepalssonl\source\repos\AspireApp1>dotnet workload install aspire

Workload(s) 'aspire' are already installed.

Successfully installed workload(s) .


C:\Users\sepalssonl\source\repos\AspireApp1>dotnet workload list

Installed Workload Id      Manifest Version      Installation Source
------------------------------------------------------------------------------
aspire                     8.1.0/8.0.100         SDK 8.0.400, VS 17.11.35327.3

Use `dotnet workload search` to find additional workloads to install.

Updates are available for the following workload(s): aspire. Run `dotnet workload update` to get the latest.
```

**Comment by LarsPal** [comment]
Obviously there have been som problems / mismatch of the Aspire workloads installed on my computer that VS reapir can't solve. Is there any other way to clear / reinstall Aspire so that I don't have to reinstall the complete computer?

**Comment by joperezr** [comment]
Sorry you are hitting issues @LarsPal. I suppose the issue you are hitting is that your SDK is trying to use workload set versions. Can you please run:

`dotnet workload --version`

That should let us know if you are using workload sets or manifests and that will help us know how to unblock you. FWIW, Aspire 9 shipped last week, and that one doesn't require a workload at all, so I'd suggest upgrading if possible. Here are some docs that can help you with that: https://learn.microsoft.com/en-us/dotnet/aspire/get-started/upgrade-to-aspire-9?pivots=visual-studio

**Comment by LarsPal** [comment]
`dotnet workload --version` returns `8.0.400-manifests.f0bb973a`.
I will discuss with my team if we can upgrade to Aspire 9.

**Comment by joperezr** [comment]
That's great to hear, thanks for confirming @LarsPal. Will close this now as the issue is resolved by moving to Aspire 9. Please let us know if you hit any issues.
